src/lib/translators/mongo.ts

```diff
@@ -290,6 +290,53 @@ function transformReplace(step: Readonly<S.ReplaceStep>): MongoStep {
   };
 }
 
+/** transform an 'replace' step into corresponding mongo steps */
+function transformRollup(step: Readonly<S.RollupStep>): MongoStep {
+  const facet: { [id: string]: MongoStep[] } = {};
+  for (const [idx, elem] of step.hierarchy.entries()) {
+    const id = columnMap([...step.hierarchy.slice(idx), ...(step.groupby ?? [])]);
+    const aggs: { [id: string]: {} } = {};
+    for (const aggfStep of step.aggregations) {
+      if (aggfStep.aggfunction === 'count') {
+        aggs[aggfStep.newcolumn] = {
+          $sum: 1,
+        };
+      } else {
+        aggs[aggfStep.newcolumn] = {
+          [$$(aggfStep.aggfunction)]: $$(aggfStep.column),
+        };
+      }
+    }
+    const project: { [id: string]: string | number } = {
+      _id: 0,
+      ...Object.fromEntries(Object.keys(id).map(col => [col, `$_id.${col}`])),
+      ...Object.fromEntries(Object.keys(aggs).map(col => [col, 1])),
+      [step.labelCol ?? 'label']: `$_id.${elem}`,
+      [step.levelCol ?? 'level']: elem,
+    };
+    if (idx < step.hierarchy.length - 1) {
+      project[step.parentCol ?? 'parent'] = `$_id.${step.hierarchy[idx + 1]}`;
+    }
+    facet[`level_${idx}`] = [
+      {
+        $group: {
+          _id: id,
+          ...aggs,
+        },
+      },
+      {
+        $project: project,
+      },
+    ];
+  }
+  return [
+    { $facet: facet },
+    { $project: { _vqbLevels: { $concatArrays: Object.keys(facet).map(col => $$(col)) } } },
```

Comment:
May I suggest to call this variable `_vqbRollupLevels` so it's easier to understand which pipeline step it comes from when reading the output query?